pkg/lightning/backend/external/writer.go

```diff
@@ -531,11 +528,16 @@ func (w *Writer) flushKVs(ctx context.Context, fromClose bool) (err error) {
 		zap.Int("sequence-number", w.currentSeq),
 	)
 	sortStart := time.Now()
-	var dupFound bool
+	var (
+		dupFound         bool
+		dupKey, dupValue []byte
+	)
 	slices.SortFunc(w.kvLocations, func(i, j membuf.SliceLocation) int {
 		res := bytes.Compare(w.getKeyByLoc(&i), w.getKeyByLoc(&j))
-		if res == 0 {
+		if res == 0 && len(dupKey) == 0 {
 			dupFound = true
+			dupKey = w.getKeyByLoc(&i)
+			dupValue = w.getValueByLoc(&i)
```

Comment:
i think we should clone it, the `KeyByLoc` is stored on a shared buffer, it's released on close